### ISSUE TITLE
Keeping Track of Previous State in State Machines

### ISSUE THREAD

**Issue by pasetti** [pnp-software/fwprofile#3]
One feedback from the CHEOPS Project is that it would often be useful to keep track of the previous state of a state machine. This would require extending the size of the State Machine Descriptor and would add a small overhead to execution of a state transition but allow, for instance, a transition action or a state entry action to query the descriptor for the source state of a transition.

**Comment by pasetti** [comment]
Marcel: do you think we should do this change? From a functional point of view, the information about the previous state would sometimes be useful but it would add a small memory and run-time overhead. The required change would be non-trivial and would require a (small) extension of the test suite.

**Comment by oppm** [comment]
Two things which come to my mind:
1. It shouldn't be much of a change in the source code, right ? I guess there is an index or pointer to a node which you simply have to copy/shadow. Ok, I agree that the test suite would need to updated. 
2. What is actually the advantage of knowing the previous state ? In my opinion, a state should never need to know the previous one, i.e. having a different behaviour depending on the previous state. If this is the case in some of the states for CHEOPS than one should maybe think about the state definitions again and introduce some additional states to cover the different behaviours. 

In summary, unless I did not really understand the reason behind this feature, I think it is wrong to implement it because if you need this feature than the states are badly defined.

**Comment by pasetti** [comment]
I agree with Marcel's assessment and I therefore close this ticket.
